system/include/emscripten/val.h

```diff
@@ -651,19 +568,37 @@ class EMBIND_VISIBILITY_DEFAULT val {
   template<typename WrapperType>
   friend val internal::wrapped_extend(const std::string& , const val& );
 
-  template<internal::EM_METHOD_CALLER_KIND Kind, typename Ret, typename Implementation, typename... Args>
-  Ret internalCall(Implementation impl, Args&&... args) const {
+  template<internal::EM_INVOKER_KIND Kind, typename Policy, typename Ret, typename... Args>
+  static Ret internalCall(EM_VAL handle, const char *methodName, Args&&... args) {
     using namespace internal;
 
+    using RetWire = BindingType<Ret>::WireType;
+
+    static constexpr typename Policy::template ArgTypeList<Ret, Args...> argTypes;
+    thread_local EM_INVOKER mc = _emval_create_invoker(argTypes.getCount(), argTypes.getTypes(), Kind);
+
     WireTypePack<Args...> argv(std::forward<Args>(args)...);
     EM_DESTRUCTORS destructors = nullptr;
-    EM_GENERIC_WIRE_TYPE result = impl(
-      Signature<Kind, Ret, Args...>::get_method_caller(),
-      as_handle(),
-      &destructors,
-      argv);
+
+    RetWire result;
+    if constexpr (std::is_integral<RetWire>::value && sizeof(RetWire) == 8) {
```

Comment:
Hmm...do we not test val with c++11? The rest of embind is c+11 compatible, though we've talked about bumping it to c++17.

Reply:
We do, and I had to fix this line a few times to make it work for C++11.

Reply:
Cool, didn't realize that it was a warning only. I've definitely wanted to use that in a few other places in embind.

Reply:
Yeah that was confusing for a moment. So are you okay with me leaving it as-is? I can refactor to use pure C++11 SFINAE I guess, but if it works, it works...

Reply:
I'm fine with you leaving it. Especially since we already have one that you pointed out already.

Reply:
Ok thanks.